dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/UnitOfWorkApplicationListener.java

```diff
@@ -73,13 +75,17 @@ private static class UnitOfWorkEventListener implements RequestEventListener {
         public void onEvent(RequestEvent event) {
             final RequestEvent.Type eventType = event.getType();
             if (eventType == RequestEvent.Type.RESOURCE_METHOD_START) {
-                methodMap
-                    .computeIfAbsent(event.getUriInfo().getMatchedResourceMethod(), UnitOfWorkEventListener::registerUnitOfWorkAnnotations)
-                    .forEach(unitOfWork ->
-                        unitOfWorkAspects
-                            .computeIfAbsent(unitOfWork.value(), hibernateName -> new UnitOfWorkAspect(sessionFactories))
-                            .beforeStart(unitOfWork)
-                );
+                try {
+                    methodMap
+                        .computeIfAbsent(event.getUriInfo().getMatchedResourceMethod(), UnitOfWorkEventListener::registerUnitOfWorkAnnotations)
+                        .forEach(unitOfWork ->
+                            unitOfWorkAspects
+                                .computeIfAbsent(unitOfWork.value(), hibernateName -> new UnitOfWorkAspect(sessionFactories))
+                                .beforeStart(unitOfWork)
+                    );
+                } catch (HibernateException e) {
```

Comment:
I think we should stay consistent here with the handling around `afterEnd` and catch all `Exception`s and not only `HibernateException`s.

Reply:
I originally did just that. However, `beforeStart` throws a diagnostic `IllegalArgumentException` if the `@UnitOfWork` being set up specifies a Hibernate bundle name that is not registered during application start up. I reasoned that errors like those are actually bugs in the application and wouldn't usually be handled through an `ExceptionMapper`. Should I wrap the `IAE` with `MapableException` anyway or should special-case it?

Reply:
We shouldn't change the current behaviour. As the `IllegalArgumentException` is the only one intentionally thrown, we can handle it as a special case.